libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1452,6 +1452,35 @@ internal object MapboxRouteLineUtils {
             }
     }
 
+    fun removeLayers(style: Style) {
+        style.removeStyleLayer(RouteLayerConstants.TOP_LEVEL_ROUTE_LINE_LAYER_ID)
+        style.removeStyleLayer(RouteLayerConstants.BOTTOM_LEVEL_ROUTE_LINE_LAYER_ID)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_1_TRAIL_CASING)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_1_TRAIL)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_1_CASING)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_1_MAIN)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_1_TRAFFIC)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_2_TRAIL_CASING)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_2_TRAIL)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_2_CASING)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_2_MAIN)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_2_TRAFFIC)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_3_TRAIL_CASING)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_3_TRAIL)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_3_CASING)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_3_MAIN)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_3_TRAFFIC)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_1_RESTRICTED)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_2_RESTRICTED)
+        style.removeStyleLayer(RouteLayerConstants.LAYER_GROUP_3_RESTRICTED)
+        style.removeStyleLayer(RouteLayerConstants.WAYPOINT_LAYER_ID)
```

Comment:
We have nicely created fields that are`setOf` different `layerIds` like `MapboxRouteLineView.trailCasingLayerIds` and others. We can move these fields as `internal` to `RouteLayerContants` and reuse them both in `MapboxRouteLineView` as well as in the above function call using `forEach`

```
fun removeLayers(style: Style) {
    RouteLayerConstants.trailCasingLayerIds.forEach { 
        style.removeStyleImage(it)
    }
    ....
    ....
}
```

Reply:
Those collections have a specific function totally unrelated to code removing the layers.

Reply:
@abhishek1508 idea leads in a good direction though - is there a way to build a single collection that we can iterate over both when adding and removing layers? Otherwise it's easy to make a mistake when a new layer is added to the list, we'd have to _remember_ to write a removal unit test as well, there's no automation that would detect the mistake for us.

Reply:
Here's an idea. According to [Map Style Specification](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#metadata), Layers have an optional `metadata` property which can hold any arbitrary data. Why not leverage that and add a `"layout-group": "arrows"` property which we could use to quickly identify which layers needs to be removed together?